### PR TITLE
misc(docs): Edits and additions for Forms and Change Detection

### PR DESCRIPTION
**Pull request by naomiblack**

### CODE REVIEW

**Comment by naomiblack** [comment]
@petebacondarwin I'm having trouble with the doc generation for the vars in `modules/angular2/change_detection.js`. We want only the vars to show up in the output, so I moved the comment description, but when I run `gulp build public_docs` I get weird output in two places.

The left nav shows the classes and vars in duplicate:
image

I expect to only see "jitChangeDetection" once, but the var is twice in the list, and the class `JitChangeDetection

**Comment by naomiblack** [comment]
I just generated docs again with master @8b97cf14797a8f01707be316d9996bd0bdae8998 and see the same duplication of the `JitChangeDetection` class in the output. So seems like a bug in either dgeni or our doc comments?

Here's what I see with @8b97cf14797a8f01707be316d9996bd0bdae8998

image

**Comment by petebacondarwin** [comment]
There were two issues here. The first is that since these classes are already in a public module they do not need to be annotated with `@exportedAs` since they are already being exported anyway. The second is that dgeni makes a clone of docs that are being re-exported, so that they can exist in both modules. I have changed this so that it doesn't clone exports which are not being exported to a different module than that in which they are originally defined.

File: modules/angular2/change_detection.js [hunk 1]
```diff
@@ -85,5 +79,16 @@ export class JitChangeDetection extends ChangeDetection {
...
+ * Implements dynamic change detection. See: [ChangeDetection] for more details.
+ *
+ * @exportedAs angular2/change_detection
+ */
 export var dynamicChangeDetection = new DynamicChangeDetection(_registry);
+
+/**
+ * Implements just-in-time change detection. See: [ChangeDetection] for more details.
+ *
+ * @exportedAs angular2/change_detection
```
Comment (petebacondarwin) [inline modules/angular2/change_detection.js]: This annotation is not needed as this var is already being exported in this public module.

**Pull request closed without merging**